presbydora.module

```diff
@@ -273,9 +285,34 @@ function presbydora_embargo_date_set_year_range($form_element, $form_values) {
  */
 function presbydora_islandora_datastream_access($op, $islandora_datastream, $user) {
   if ($islandora_datastream->id == 'OBJ' || $islandora_datastream->id == 'PDF') {
+    // Modified to allow specific objects to be accessible.
+    $object = $islandora_datastream->parent;
+    $pid = $object->id;
+
+    module_load_include('inc', 'presbydora', 'includes/utilities');
+    $bypass_pids = variable_get('presbydora_access_bypass_pids', array());
+    if (!empty($bypass_pids)) {
+      $ancestors = presbydora_get_ancestors($pid);
+      $is_intersect = !empty(array_intersect($ancestors, $bypass_pids));
+
+      if ($is_intersect) {
+        return NULL;
+      }
+    }
     $guest = user_role_load_by_name('PHS Guest');
     if ($user->uid == 0 || (isset($guest->rid) && user_has_role($guest->rid, $user))) {
       return FALSE;
     }
   }
 }
+
+/**
+ * Implements hook_cmodel_pid_islandora_object_purged().
+ */
+function presbydora_islandora_collectioncmodel_islandora_object_purged($pid) {
+  $bypass_pids = variable_get('presbydora_access_bypass_pids', array());
+  if (!empty($bypass_pids)) {
+    unset($bypass_pids[$pid]);
+    variable_set('presbydora_access_bypass_pids', $bypass_pids);
```

Comment:
We _could_ avoid the `variable_set()` here, if we didn't actually `unset()` anything; though, not really concerned.

Reply:
Changed to isset over !empty